python/tank/platform/environment.py

```diff
@@ -356,44 +356,55 @@ def find_location_for_engine(self, engine_name):
                             % (engine_name, self._env_path))
             
         return tokens, path
-    
+
+    def is_framework_available_from(self, framework_name, starting_point):
+        """
+        Tests if a framework is reachable from a given file in the environment by following its includes.
+
+        :param framework_name: Name of the framework to search for.
```

Comment:
i always get confused about the instance name and the fw name - could we add an example to the docstring? Which one is it in this case? 'tk-framework-widget_v0.2.x' or 'tk-framework-widget'?

```
:param framework_name: Name of the framework to search for, e.g. 'tk-framework-shotgunutils'
```